lib/tasks/recalculate_vacdays.rake

```diff
@@ -0,0 +1,5 @@
+desc "Recalculate vacdays after bugfix for daylight savings time changes"
+task recalculate_vacdays: :environment do
+  logs = LettingsLog.filter_by_years(%w[2023 2024]).where.not(vacdays: nil)
+  logs.each(&:save!)
+end
```

Comment:
Mainly a task for ease of kicking it off on an ad hoc instance.

This is filtered to open years to avoid validation issues - need to check with the stats team what to do about older data. The export after running this can be expected to be relatively large, but that should be ok.

Reply:
I hope we don't have invalid logs on prod, but in case there are any, I think save! would fail with an error and stop processing?

Also, would it make sense to only save where the value would have changed to avoid the large export? I assume some of the values were correct?

Reply:
That's a good point - I'd expect nothing to be invalid but it could be if the validations have changed since it was touched. I guess maybe we don't actually care about whether it's valid for this purpose, although it would be nice to find that out.

We need to run `set_derived_fields!` which is in a before validation hook, but we could instead do that manually and not validate.

On the value changing, I think that rails will only actually make a database change (and thus update the updated_at time etc) if an attribute has actually been changed - will double check this.

Reply:
Ok, I'm trying something that will just manually update the vacdays value (and not run the validations so that it can work on all logs regardless of collection year - this also means that if there are currently invalid logs it will still work, which seems not ideal but also it's not making it worse).

Reply:
You could potentially run `unless log.save Rails.logger.something_useful` as well